libp2p/connmanager.nim

```diff
@@ -53,28 +58,42 @@ type
     muxer: Muxer
     handle: Future[void]
 
+  ConnHolder = object
+    conn: Connection
+    evtHandle: Future[void]
```

Comment:
re names, I think what I intended here more precisely is that `handle` is/can a passive noun, but what's being held in the variable is an active ongoing task effectively, so an active noun/ verb would be more appropriate

Reply:
OK, my reasoning is that this is a handle for some opaque resource, in this case it is the handle for the close event and as such, unline with a read loop for example, there isn't an ongoing action associated with it. I'm fine changing it, but I struggle to come up with a more appropriate name - maybe `cancelHandle`?

Reply:
err, `onCloseHandle`?

Reply:
ah right, I think I confused it with other places where `handle` is used for the read loop - but even in this case, there is an "activity" associated with the handle - an ongoing `onClose` call which is stuck in `join` - and in fact, this is a leak as well: when the connection is removed from `conns`, `onClose` is not cancelled - this means that the future and everything it owns will never get released

Reply:
ping, leak?

Reply:
Well, the original flow hasn't changed, everywhere the connection is being removed there is a `conn.close()` in the vicinity (https://github.com/status-im/nim-libp2p/blob/master/libp2p/connmanager.nim#L217-L223, https://github.com/status-im/nim-libp2p/blob/master/libp2p/connmanager.nim#L374-L387), so that flow is the same, so it doesn't look like it would leak ATM.